Dockerfile

```diff
@@ -1,23 +1,48 @@
-FROM ubuntu:20.04
+# --------------------------------------------------
+# Base container
+# --------------------------------------------------
+FROM docker.io/ubuntu:focal AS base
 
 RUN set -e
 
-# Setup time-zone so that the build does not hang
-# on configuring the tzdata package.
-# I work in Brno, that is basically Vienna-North :)
-# (definitely its closer than Prague)
-ENV TZ=Europe/Vienna
-RUN ln -snf "/usr/share/zoneinfo/$TZ" /etc/localtime
-RUN echo "$TZ" > /etc/timezone
-
 # Install packages
+ENV DEBIAN_FRONTEND=noninteractive
 RUN apt-get update
-RUN apt-get install -y git cmake make llvm zlib1g-dev clang g++ python3
+RUN apt-get install -yq --no-install-recommends clang llvm
+
+# --------------------------------------------------
+# Build container
+# --------------------------------------------------
+FROM base as build
+
+# Install build dependencies
+RUN apt-get install -yq --no-install-recommends ca-certificates cmake git \
+                                                ninja-build llvm-dev python3
 
+# Clone
 WORKDIR /opt
 RUN git clone https://github.com/mchalupa/dg
+
+# Build
 WORKDIR /opt/dg
-RUN cmake . -DCMAKE_INSTALL_PREFIX=/usr
-RUN make -j2
-RUN make check -j2
-RUN make install
+
+# libfuzzer does not like the container environment
+RUN cmake -S. -GNinja -Bbuild -DCMAKE_INSTALL_PREFIX=/dg \
```

Comment:
By the way, is there any reason to install DG to a non-standard path (e.g. `/usr` or `/opt`)?

Reply:
Oh, probably because of the `COPY` command later, right??

Reply:
Yeah, with `\usr` I'd have to `COPY` everything one-by-one to the new container and I can change `\dg` to `\opt\dg`, if you'd like me to, it's true that the current solutions does not comply with the Filesystem Hierarchy Standard.

Reply:
>  I can change `\dg` to `\opt\dg`, if you'd like me to

Yeah, actually I would like that (or rather to `/opt/dg` ;) ). Sorry that I did catch this earlier.

Reply:
Done! :)